edalize/templates/yosys/yosys-script-tcl.j2

```diff
@@ -26,6 +26,11 @@ verilog_defaults -add {{ incdirs }}
 verilog_defaults -pop
 
 {{ synth_command }} {{ synth_options }} -top {{ top }}
+
+{% for command in additional_commands %}
+{{ command }}
+{% endfor %}
+
```

Comment:
We already have the tclSource file type which allows us to inject commands. However, those commands will run before synthesis and it looks like you need to run these additional commands after synthesis. I wonder if we can come up with some clever way here. I have two ideas, but I'm not sure if either of them are very good.
1. Add support for defining custom attributes to files in the EDAM file. This would e.g. allow us to tag which tcl scripts to run before and after simulation
2. Define a post-synthesis hook in the main yosys script. This would allow users to register their own functions to be run after synthesis

The first alternative is more generic and I have been thinking along these lines for other use cases. However, it will require a fair amount of work in Edalize (and FuseSoC). The second one will make tcl scripts that use this hook Edalize-specific, but I'm still inclined to go this way.

Or perhaps it's best to just do it the way you propose. I'm just a bit careful about adding additional mechanisms for injecting commands

Reply:
You know what, I think the best long-term solution is to allow replacing the script template with a user-defined one. Yosys is becoming this swiss-army knife and I want to support as many reasonable use-cases as possible. So before we can merge this PR I want to do that work on yosys. I'll try to get it done ASAP.

Reply:
I'm not sure I understand. Is it the paths to the json and xdc files you want to parameterize? If so I see three ways to do that. Either supply a TCL template file that reads some env vars, generate the template with correct paths before handing it to Edalize or make the TCL template call something externally that provides this information. Would any of those alternatives work. Not really sure how parameterization would work otherwise.

The exception is if you need some design-specific parameters but in that case I think we should make those available as variables, like I did with $top

Reply:
Thanks for suggestions. I tested the approach with env vars and it works for me. I'll implement `yosys_template` solution to this PR.

Reply:
I've reworked the code to support Yosys templates. I will rebase the branch when the `yosys_template` branch will be merged to master.

Reply:
Great. Thanks for trying it out